src/main/java/com/cognifide/aemrules/checks/ContentResourceShouldBeNullCheckedCheck.java

```diff
@@ -0,0 +1,266 @@
+        if (isRightSideNullCheck(tree)) {
+            contentResources.replace(tree.condition().firstToken().text(), value);
+        } else {
+            contentResources.replace(tree.condition().lastToken().text(), value);
+        }
+    }
+
+    private void externalLibraryNullChecks(MethodInvocationTree tree) {
+        if (isNonNullUsed(tree)) {
+            contentResources.replace(tree.arguments().get(0).toString(), true);
+        } else if (isAllNonNullUsed(tree)) {
+            tree.arguments().forEach(contentResource -> contentResources.replace(contentResource.toString(), true));
+        } else if (insideIfStatement && isThisIsNullMethod(tree)) {
+            insideEqualNullCheck = true;
+        }
+    }
+
+    private void chainedMethodsNullCheck(MethodInvocationTree tree) {
+        if (tree.methodSelect() instanceof MemberSelectExpressionTree) {
+            MemberSelectExpressionTree method = (MemberSelectExpressionTree) tree.methodSelect();
+            if (method.expression() instanceof MethodInvocationTree) {
+                MethodInvocationTree invocation = (MethodInvocationTree) method.expression();
+                if (!invocation.symbol().isUnknown() &&
+                    isPage(invocation.symbol().owner().type().fullyQualifiedName()) &&
+                    isGetContentResourceUsedOnPage(invocation) && !returnOccurredInsideEqualNullCheck
+                ) {
+                    context.reportIssue(this, tree, RULE_MESSAGE);
+                }
+            }
+        }
+    }
+
+    private boolean isResource(AssignmentExpressionTree tree) {
+        return Constants.RESOURCE_TYPE.equals(tree.variable().symbolType().fullyQualifiedName());
+    }
+
+    private boolean isResourceInitialized(VariableTree tree) {
+        return tree.initializer() != null && tree.initializer().kind() != Kind.NULL_LITERAL;
+    }
+
+    private boolean isGetContentResourceUsedOnPage(MethodInvocationTree tree) {
+        return METHOD_NAME.equals(tree.methodSelect().lastToken().text());
+    }
+
+    private boolean isGetContentResourceUsedOnPage(VariableTree tree) {
+        return tree.initializer() instanceof MethodInvocationTree &&
+            !((MethodInvocationTree) tree.initializer()).symbol().isUnknown() &&
+            isPage(((MethodInvocationTree) tree.initializer()).symbol().owner().type()
+                .fullyQualifiedName()) &&
+            METHOD_NAME.equals(((MethodInvocationTree) tree.initializer()).symbol().name());
+    }
+
+    private boolean isGetContentResourceUsedOnResource(AssignmentExpressionTree tree) {
+        return isResource(tree) &&
+            tree.expression() instanceof MethodInvocationTree &&
+            METHOD_NAME.equals(((MethodInvocationTree) tree.expression()).symbol().name());
+    }
+
+    private boolean isPage(String name) {
+        return Constants.PAGE.equals(name);
+    }
+
+    private boolean isThisAResourceNullCheck(IfStatementTree tree, String ifType) {
```

Comment:
Please introduce result variable
boolean result = false
and do not use return in if statement

Reply:
done